components/service/glean/src/main/java/mozilla/components/service/glean/Dispatchers.kt

```diff
@@ -64,6 +85,61 @@ internal object Dispatchers {
         fun setTestingMode(enabled: Boolean) {
             testingMode = enabled
         }
+
+        /**
+         * Stops queueing tasks and processes any tasks in the queue
+         */
+        fun flushQueuedInitialTasks() {
+            // Setting this to false first should cause any new tasks to just be executed (see
+            // launch() above) making it safer to process the queue.
+            queueInitialTasks = false
+
+            for (task in taskQueue) {
+                task.invoke()
+            }
+            taskQueue.clear()
+        }
+
+        /**
+         * Helper function to add task to queue as either a synchronous or asynchronous operation,
+         * depending on whether [testingMode] is true.
+         */
+        private fun addTaskToQueue(block: suspend CoroutineScope.() -> Unit) {
+            if (testingMode) {
+                logger.info("Task queued for execution in test mode")
+                taskQueue.add {
+                    runBlocking {
+                        block()
+                    }
+                }
+            } else {
+                if (taskQueue.size >= MAX_QUEUE_SIZE) {
```

Comment:
I think this should be the first check of this function: the cap should be there regardless of the mode we're in. This would simplify analysis in case there's any bug around this.

```
if (taskQueue.size >= MAX_QUEUE_SIZE) {
  logger.error("Exceeded maximum queue size, discarding task")
  return
}

if (testingMode) {
    logger.info(...)
    taskQueue.add {
      block()
     }
} else {
    logger.info("Task queued for execution and delayed until flushed")
    taskQueue.add {
      coroutineScope.launch(block = block)
    }
}
```